### PR TITLE
Mask libvirtd sockets which prevents cloudstack-agent from being setup

### PR DESCRIPTION
**Pull request by Pearl1594**
### Description

This PR fixes the issue of libvirtd service not starting successfully, which leads to failure in setting up of cloudstack-agent
Addresses: https://github.com/apache/cloudstack/issues/4754

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [X] Major
- [ ] Minor
- [ ] Trivial

### How Has This Been Tested:
Issue noticed when cloudstack-setup-agent is run (while adding hosts):
```
[root@ref-trl-2525-k-M8-pearl-dsilva-kvm1 ~]# cloudstack-setup-agent  -m XX.XX.XX.XX -z 1 -p 1 -c 1 -g 4fe04fea-515c-4c4c-9c78-798554b483b2 -a -s  --pubNic=cloudbr1 --prvNic=cloudbr0 --guestNic=cloudbr1 --hypervisor=kvm
Starting to configure your system:
Configure SElinux ...         [OK]
Configure Network ...         [OK]
Configure Libvirt ...         [Failed]
Configuration failed for service Libvirt
Try to restore your system:
Restore SElinux ...           [OK]
Restore Network ...           [OK]
Restore Libvirt ...           [Failed]

```
Post Fix:
```
[root@ref-trl-2528-k-M8-pearl-dsilva-kvm1 ~]# cloudstack-setup-agent  -m XX.XX.XX.XX -z 1 -p 1 -c 1 -g bd24ab4a-4311-357e-8e9e-dab829448772 -a -s  --pubNic=cloudbr1 --prvNic=cloudbr0 --guestNic=cloudbr1 --hypervisor=kvm
Starting to configure your system:
Configure SElinux ...         [OK]
Configure Network ...         [OK]
Configure Libvirt ...         [OK]
Configure Firewall ...        [OK]
Configure Nfs ...             [OK]
Configure cloudAgent ...      [OK]
CloudStack Agent setup is done!

```

### CODE REVIEW

**Comment by Pearl1594** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@Pearl1594 I think -l works with libvirtd6 as well, on Ubuntu 20.04 you need to mask few things I think. (https://rohityadav.cloud/blog/cloudstack-rpi4-kvm/#kvm-host-setup)
If need this fix, can you change PR base branch to 4.15?

**Comment by Pearl1594** [comment]
@rhtyd -l doesn't seem to be working with libvirt 6.0.0, following is the exception that is noticed:
```
  4 09:15:09 ref-trl-2528-k-M8-pearl-dsilva-kvm1 journal[4103]: libvirt version: 6.0.0, package: 28.module_el8.3.0+555+a55c8938 (CentOS Buildsys <bugs@centos.org>, 2020-11-04-01:04:00, )
Mar  4 09:15:09 ref-trl-2528-k-M8-pearl-dsilva-kvm1 journal[4103]: --listen parameter not permitted with systemd activation sockets, see 'man libvirtd' for further guidance
Mar  4 09:15:09 ref-trl-2528-k-M8-pearl-dsilva-kvm1 systemd[1]: libvirtd.service: Main process exited, code=exited, status=6/NOTCONFIGURED
Mar  4 09:15:09 ref-trl-2528-k-M8-pearl-dsilva-kvm1 systemd[1]: libvirtd.service: Failed with result 'exit-code'.

```
I've rebased the PR off 4.15

**Comment by Pearl1594** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@Pearl1594 the errors I think are hinting towards disabling the sockets:

systemctl mask libvirtd.socket libvirtd-ro.socket libvirtd-admin.socket libvirtd-tls.socket libvirtd-tcp.socket
systemctl restart libvirtd

**Comment by rhtyd** [comment]
Check the code wrt Ubuntu, I think the scripts do this for Ubuntu 20.04 already @Pearl1594

**Comment by Pearl1594** [comment]
Thanks @rhtyd - I'll take that as reference and align to it.

**Comment by Pearl1594** [comment]
@blueorangutan package

File: python/lib/cloudutils/serviceConfig.py [hunk 1]
```diff
@@ -22,6 +22,8 @@
 from .configFileOps import configFileOps
 import os
 import shutil
+from distutils.version import StrictVersion
+
```
Comment (shwstppr) [inline python/lib/cloudutils/serviceConfig.py:26]: @Pearl1594 I guess not needed anymore

**Comment by weizhouapache** [comment]
> Check the code wrt Ubuntu, I think the scripts do this for Ubuntu 20.04 already @Pearl1594

yep. I added the lines.
ubuntu 20.04 + cloudstack 4.15 works fine.

**Comment by DaanHoogland** [comment]
@blueorangutan test

**Review by rhtyd — APPROVED** [review]

**Comment by rhtyd** [comment]
@blueorangutan test centos8 kvm-centos8

**Review by shwstppr — APPROVED** [review]
Code changes LGTM.
OOBM tests failing with `Unable to Get Channel Cipher Suites\nSet Session Privilege Level to ADMINISTRATOR failed\nError: Unable to establish IPMI v2 / RMCP+ session\n`. This need to be checked

**Review by rhtyd — APPROVED** [review]

**Pull request merged**